database/src/lib.rs

```diff
@@ -798,3 +798,154 @@ pub struct ArtifactCollection {
     pub duration: Duration,
     pub end_time: DateTime<Utc>,
 }
+
+#[derive(Debug)]
+pub enum BenchmarkRequestCommitType {
+    Try { pr: u32 },
+    Master { pr: u32 },
+    Release { tag: String },
+}
+
+impl fmt::Display for BenchmarkRequestCommitType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BenchmarkRequestCommitType::Try { pr: _ } => write!(f, "try"),
+            BenchmarkRequestCommitType::Master { pr: _ } => write!(f, "master"),
+            BenchmarkRequestCommitType::Release { tag: _ } => write!(f, "release"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum BenchmarkRequestStatus {
+    WaitingForArtifacts,
+    WaitingForParent,
+    Queued,
+    InProgress,
+    Completed,
+}
+
+impl fmt::Display for BenchmarkRequestStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BenchmarkRequestStatus::WaitingForArtifacts => write!(f, "waiting_for_artifacts"),
+            BenchmarkRequestStatus::WaitingForParent => write!(f, "waiting_for_parent"),
+            BenchmarkRequestStatus::Queued => write!(f, "queued"),
```

Comment:
I think that queued and in progress are mostly the same thing, as per the document? Maybe `Queued` is a better name though. For the purpose of the benchmark request, it shouldn't matter much if the request is only queued or actually running.